apps/rule-manager/client/src/ts/components/hooks/useRule.ts

```diff
@@ -116,6 +117,59 @@ export function useRule(ruleId: number | undefined) {
     }
   }
 
+  const updateRule = async (ruleForm: DraftRule) => {
```

Comment:
Consolidating update and create operations in this effect lets them share error and loading states, which makes our UI more consistent (loading state now accounts for fetching, creating, and updating, for example.)